src/algebra/monoid_algebra.lean

```diff
@@ -1177,6 +1213,47 @@ See note [partially-applied ext lemmas]. -/
   f = g :=
 ring_hom_ext (ring_hom.congr_fun h₁) (monoid_hom.congr_fun h_of)
 
+section opposite
+
+open finsupp opposite
+
+variables [semiring k]
+
+/-- The opposite of an `add_monoid_algebra R I` is additively equivalent to
+the `add_monoid_algebra Rᵒᵖ I` over the opposite ring, taking elements to their opposite. -/
+@[simps apply symm_apply] protected noncomputable def op_add_equiv :
+  (add_monoid_algebra k G)ᵒᵖ ≃+ add_monoid_algebra kᵒᵖ G :=
+opposite.op_add_equiv.symm.trans (finsupp.map_range.add_equiv opposite.op_add_equiv)
+
+lemma op_add_equiv_single (i : G) (r : k):
+  add_monoid_algebra.op_add_equiv (op (single i r)) = single i (op r) :=
+by simp
+
+/-- The opposite of an `add_monoid_algebra R I` is ring equivalent to
+the `add_monoid_algebra Rᵒᵖ I` over the opposite ring, taking elements to their opposite.
+For the case where the index type `I` is not `[add_comm_monoid I]`, see
+`add_monoid_algebra.op_add_equiv`. -/
+@[simps {simp_rhs := tt}] protected noncomputable def op_ring_equiv [add_comm_monoid G] :
+  (add_monoid_algebra k G)ᵒᵖ ≃+* add_monoid_algebra kᵒᵖ G :=
+{ map_mul' := begin
+    dsimp only [add_equiv.to_fun_eq_coe, ←add_equiv.coe_to_add_monoid_hom],
+    rw add_monoid_hom.map_mul_iff,
+    ext i r i' r' : 6,
+    dsimp,
+    simp only [map_range_single, single_mul_single, ←op_mul, add_comm]
+  end,
+  ..add_monoid_algebra.op_add_equiv }
```

Comment:
I'd lean towards removing this too, and using
```suggestion
  .. opposite.op_add_equiv.symm.trans (finsupp.map_range.add_equiv opposite.op_add_equiv) }
```

Reply:
Why's that? Because it's unlikely that one will have the algebra defined over an arbitrary index type that isn't an add_comm_monoid?

Reply:
There are all sorts of ways someone might want to apply `op` to parts of a type, but most of them aren't interesting. The thing that makes this one interesting is that it preserves multiplication.  The additive-only version has no such redeeming feature.